chore: generate release notes and update changelog

davidecavestro opened the pull request:
Add some logic just after existing release management, in order to:
- gather commits included (commit after last release/tag)
- replace current release description with the list of released commits (grouped by conventional commit type)
- update _Changelog.md_ accordingly on _master_ branch

Please note it can work just if commit messages comply with https://www.conventionalcommits.org/en/v1.0.0/

rogosprojects commented:
It sounds overall a good improvement.
I think it will be better if ae create at first a  "draft release" (nobody will be email notified with initial release note), then updating it with the right content of release (note + artifacts)

```     - name: Create Draft Release
        uses: ncipollo/release-action@v1.14.0
        with:
          prerelease: false
          draft: true
          commit: ${{ github.sha }}
          body: '*pending*' 
```

Or, creating a release after gathering facts ( release body , artifacts ).

davidecavestro commented:
> I think it will be better if we create at first a "draft release"

It is done this way just to avoid breaking something on the release actions currently in place, as there's both
_Roang-zero1/github-create-release-action_ and
_wangyoucao577/go-release-action_

Since I couldn't figure out the motivation behind the choice to use 2 actions, I tried not to mess with them and just add some post-processing.

I suspected it was someway related to somw async behaviour of the build matrix, but I'd also say that any step starts just after previous one completed.

That said, I'm more than happy if you agree to make things simpler, possibly avoiding also to use a draft release.

IMHO the best sequence would be
- checkout
- detect changes
- build artifacts
- create the release attaching artifacts and setting body from detected changes
- OPTIONAL commit changes to changelog.md

> Also, keep in mind that the actual flow will get the release body from`CHANGELOG.md` file. This will be useless since the Action will write it then

Nope. It uses [the output](https://github.com/requarks/changelog-action?tab=readme-ov-file#outputs) of the action that generates the markdown for cumulated changes since last release/tag.
This output is also prepended to the Changelog.md, and this can be turned off setting `writeToFile: false`
Then the changes to _Changelog.md_ are committed to master, supposing this is something useful, but - again - it is entirely optional.

rogosprojects commented:
I agree with you, let's simplify!
So, i'm going to merge it and testing in this repository.

👀 Since you're a contributor, feel free to create a branch for next improvements so we can test here.

rogosprojects commented:
Awesome:
- https://github.com/rogosprojects/klogs/commit/49530cf643971506b2f9b6f905bec7f34c2997db
- https://github.com/rogosprojects/klogs/releases/tag/v1.1.15
